lib/publiccloud/eks.pm

```diff
@@ -9,43 +9,24 @@
 # Documentation: https://docs.aws.amazon.com/AmazonECR/latest/userguide/docker-push-ecr-image.html
 
 package publiccloud::eks;
-use Mojo::Base 'publiccloud::aws';
+use Mojo::Base 'publiccloud::k8s_provider';
 use testapi;
 use utils;
 
-has security_token => undef;
-
-sub vault_create_credentials {
-    my ($self) = @_;
-
-    record_info('INFO', 'Get credentials from VAULT server for EKS');
-    my $path = '/aws/sts/openqa-role-eks';
-    my $res = $self->vault->api('/v1/' . get_required_var('PUBLIC_CLOUD_VAULT_NAMESPACE') . $path, method => 'post');
-    my $data = $res->{data};
-
-    $self->key_id($data->{access_key});
-    $self->key_secret($data->{secret_key});
-    $self->security_token($data->{security_token});
-    die('Failed to retrieve key')
-      unless (defined($self->key_id) && defined($self->key_secret) && defined($self->security_token));
-
-    assert_script_run('export AWS_SESSION_TOKEN="' . $self->security_token . '"');
-    assert_script_run("export AWS_ACCESS_KEY_ID=" . $self->key_id);
-    assert_script_run("export AWS_SECRET_ACCESS_KEY=" . $self->key_secret);
-    assert_script_run('export AWS_DEFAULT_REGION="' . $self->region . '"');
-}
-
-sub _check_credentials {
-    my ($self) = @_;
-    return 1;
-}
+has provider_client => undef;
 
 sub init {
     my ($self, %args) = @_;
-    $self->SUPER::init();
+    $self->SUPER::init("EKS");
```

Comment:
see comment below, maybe it would be better to call it something like
```
    $self->SUPER::init(publiccloud::aws_client->new(
                region => $self->region,
                service => $service
            ));
```